src/pages/signin.tsx

```diff
@@ -34,12 +39,31 @@ export default function Signin() {
   });
 
   const [serverErrorState, setServerError] = useState<string | null>(null);
+  const dispatch = useAppDispatch();
+  const currentUserid = useAppSelector(selectUserid);
+  const router = useRouter();
 
-  const onSubmitHandler: SubmitHandler<FormValues> = async () => {
+  const onSubmitHandler: SubmitHandler<FormValues> = async (formData) => {
```

Comment:
You can destructure the values right here

```suggestion
  const onSubmitHandler: SubmitHandler<FormValues> = async ({ email, password }) => {
```